homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,219 @@
+
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    sensor = HVVDepartureSensor(hass, config_entry, session)
+    async_add_devices([sensor], True)
+
+
+class HVVDepartureSensor(Entity):
+    """HVVDepartureSensor class."""
+
+    def __init__(self, hass, config_entry, session):
+        """Initialize."""
+        self.config_entry = config_entry
+        self.station_name = self.config_entry.data["station"]["name"]
+        self.attr = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._available = False
+        self._state = None
+        self._name = f"Departures at {self.station_name}"
+        self._last_error = None
+
+        self.gti = GTI(
+            Auth(
+                session,
+                self.config_entry.data[CONF_USERNAME],
+                self.config_entry.data[CONF_PASSWORD],
+                self.config_entry.data[CONF_HOST],
+            )
+        )
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self, **kwargs):
+        """Update the sensor."""
+
+        departure_time = utcnow() + timedelta(
+            minutes=self.config_entry.options.get("offset", 0)
+        )
+
+        payload = {
+            "station": self.config_entry.data["station"],
+            "time": {
+                "date": departure_time.strftime("%d.%m.%Y"),
+                "time": departure_time.strftime("%H:%M"),
+            },
+            "maxList": MAX_LIST,
+            "maxTimeOffset": MAX_TIME_OFFSET,
+            "useRealtime": self.config_entry.options.get("realtime", False),
+        }
+
+        if "filter" in self.config_entry.options:
+            payload.update({"filter": self.config_entry.options["filter"]})
+
+        try:
+            data = await self.gti.departureList(payload)
+        except InvalidAuth as error:
+            if self._last_error != InvalidAuth:
+                _LOGGER.error("Authentication failed: %r", error)
+                self._last_error = InvalidAuth
+            self._available = False
+        except ClientConnectorError as error:
+            if self._last_error != ClientConnectorError:
+                _LOGGER.warning("Network unavailable: %r", error)
+                self._last_error = ClientConnectorError
+            self._available = False
+        except Exception as error:  # pylint: disable=broad-except
+            if self._last_error != error:
+                _LOGGER.error("Error occurred while fetching data: %r", error)
+                self._last_error = error
+            self._available = False
+
+        if data["returnCode"] == "OK" and data.get("departures"):
+
+            if self._last_error == ClientConnectorError:
+                _LOGGER.debug("Network available again")
+
+            self._last_error = None
+
+            departure = data["departures"][0]
+            line = departure["line"]
+            delay = departure.get("delay", 0)
+            self._available = True
+            self._state = (
+                departure_time
```

Comment:
The state for timestamp sensors should use absolute UTC iso-formatted time. Is that what this is? It doesn't seem like it's iso-formatted.

Reply:
I'm pretty sure it is I'm even using a HA method: from homeassistant.util.dt import utcnow
utcnow()

2020-06-08T16:17:04.397498+00:00 which is a valid iso time and HA can display it correctly

Reply:
We want an isoformatted string representing the datetime.
```
Python 3.8.2 (default, Feb 26 2020, 02:56:10) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from homeassistant.util.dt import utcnow                                                                                                                                                              

In [2]: utcnow()                                                                                                                                                                                              
Out[2]: datetime.datetime(2020, 6, 8, 18, 2, 2, 823844, tzinfo=<UTC>)

In [3]: utcnow().isoformat()                                                                                                                                                                                  
Out[3]: '2020-06-08T18:02:05.869791+00:00'
```